File: Lab/Tp1/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool isEnd(char *str)
{
      return strcmp(str, "FIM") == 0;
}

bool isPalindromo(char *str)
{
      int i = 0;
      int j = strlen(str) - 1;

      while (i < j)
      {
            if (str[i] != str[j])
            {
                  printf("NAO\n");
                  return false;
            }
            i++;
            j--;
      }
      printf("SIM\n");
      return true;
}

int main()
{
      char str[1000];
      do
      {
            fgets(str, 1000, stdin);
            strtok(str, "\n");
            if (!isEnd(str))
            {
                  isPalindromo(str);
            }
      } while (!isEnd(str));
      return 0;
}

File: Lab/Tp1/Palindromo.java
import java.util.Scanner;

public class Palindromo {

      public static boolean isEnd(String str) {
            return str.equals("FIM");
      }

      public static boolean isPalindromo(String str) {
            int i = 0;
            int j = str.length() - 1;

            while (i < j) {
                  if (str.charAt(i) != str.charAt(j)) {
                        System.out.println("NAO");
                        return false;
                  }
                  i++;
                  j--;
            }
            System.out.println("SIM");
            return true;
      }

      public static void main(String[] args) {
            Scanner scanner = new Scanner(System.in);
            String str;
            do {
                  str = scanner.nextLine();
                  if (!isEnd(str)) {
                        isPalindromo(str);
                  }
            } while (!isEnd(str));
            scanner.close();
      }
}
